src/gui/sdlmain.cpp

```diff
@@ -781,30 +840,27 @@ static bool LoadGLShaders(const char *src, GLuint *vertex, GLuint *fragment) {
 /*       window resolution, therefore: consider a special case for PP mode. */
 static void GetAvailableArea(int &width, int &height)
 {
-	bool   fixed;
-	double par;
+	bool fixed = false;
 
-	fixed = false;
 	if (sdl.desktop.fullscreen) {
 		if (sdl.desktop.full.fixed) {
 			width  = sdl.desktop.full.width;
 			height = sdl.desktop.full.height;
 			fixed   = true;
 		}
-	}
-	else {
+	} else {
 		if (sdl.desktop.window.width > 0) {
 			width  = sdl.desktop.window.width;
 			height = sdl.desktop.window.height;
 			fixed   = true;
 		}
 	}
 	if (!fixed) {
-		par = sdl.draw.pixel_aspect;
+		const double par = sdl.draw.pixel_aspect;
 		if (par > 1.0)
-			height = static_cast<int>(round(height * par));
+			height = iround(height * par);
 		if (par < 1.0)
-			width  = static_cast<int>(round(width  / par)); 
+			width = iround(width / par);
 	}
```

Comment:
Is it OK that either `width` or `height` are not set, depending which branch we go down?

Reply:
Yeah, this code comes from pixel-perfect implementation and we raised the same issue during review back then… I will rework the function interface to make the problem disappear.

Reply:
Addressed in 934682a50448f27d009fbed9880b37ca8cc9bfca